xgi/generators/nonuniform.py

```diff
@@ -288,3 +290,51 @@ def dcsbm_hypergraph(k1, k2, g1, g2, omega):
 
     df = pd.DataFrame(bipartite_edges)
     return Hypergraph(df)
+
+
+def random_hypergraph(N, ps):
+    """Generates a random hypergraph
+
+    Generate N nodes, and connect any d+1 nodes
+    by a hyperedge with probability ps[d-1].
+
+    Parameters
+    ----------
+    N : int
+        Number of nodes
+    ps : list of float
+        List of probabilities (between 0 and 1) to create a
+        hyperedge at each order d between any d+1 nodes. For example,
+        ps[0] is the wiring probability of any edge (2 nodes), ps[1]
+        of any triangles (3 nodes).
+
+    Returns
+    -------
+    Hypergraph object
+        The generated hypergraph
+
+    """
+
+    if (np.any(np.array(ps) < 0)) or (np.any(np.array(ps) > 1)):
+        raise ValueError("All elements of ps must be between 0 and 1 included.")
+
+    # I first generate a standard ER graph with edges connected with probability p1
+    G = nx.fast_gnp_random_graph(N, ps[0], seed=None)
+
+    nodes = list(G.nodes())
+    hyperedges = list(G.edges())
+
+    for i, p in enumerate(ps[1:]):
+        d = i + 2  # order (+2 because we started with [1:])
+
+        for hyperedge in combinations(nodes, d + 1):
```

Comment:
It looks like you didn't import combinations or NetworkX. Another thing (in the future) is that for sparse hypergraphs, it probably is faster to write a function that goes from an index to a tuple (Similar to line 27 in https://github.com/nwlandry/hypergraph-assortativity/blob/main/UNUSED/GenerativeModels.py)